src/Extensions/Subsites/IndexConfigurationExtension.php

```diff
@@ -4,23 +4,63 @@
 
 use SilverStripe\Core\Extension;
 use SilverStripe\SearchService\DataObject\DataObjectDocument;
+use SilverStripe\SearchService\Interfaces\DataObjectDocumentInterface;
 use SilverStripe\SearchService\Interfaces\DocumentInterface;
 
 class IndexConfigurationExtension extends Extension
 {
 
     public function updateIndexesForDocument(DocumentInterface $doc, array &$indexes): void
     {
-        $docSubsiteId = null;
+        // Skip if document object does not implement DataObject interface
+        if (!$doc instanceof DataObjectDocumentInterface) {
+            return;
+        }
+
+        $docSubsiteId = 0;
 
         if ($doc instanceof DataObjectDocument) {
-            $docSubsiteId = $doc->getDataObject()->SubsiteID ?? null;
+            $docSubsiteId = $doc->getDataObject()->SubsiteID ?? 0;
+        }
+
+        $this->updateDocumentWithoutSubsite($doc, $indexes, (int)$docSubsiteId);
```

Comment:
Think this makes more sense as an `if` rather than two early-returning functions. Basically
```php

$docSubsiteId = (int) $doc->getDataObject()->SubsiteID ?? 0;
if ($docSubsiteId === 0) {
  $this->updateDocumentWithoutSubsite($doc, $indexes);
} else {
  $this->updateDocumentWithSubsite($doc, $indexes, $docSubsiteId);
}
```